js/stores/__tests__/message_store_test.js

```diff
@@ -0,0 +1,37 @@
+jest.dontMock('../message_store');
+jest.dontMock('object-assign');
+jest.dontMock('react/lib/keyMirror');
+
+describe('MessageStore', function() {
+
+  var AppDispatcher;
+  var MessageStore;
+  var callback;
+  
+  var ActionTypes = require('../../constants/constants').ActionTypes;
+
+  var actionCreateMessage = {
+    type: ActionTypes.NEW_MESSAGE,
+    data: {
+      user: "Bob",
+      text: "Hello, everyone I'm new.",
+      timestamp: new Date().getTime()
+    }
+  };
+
+  beforeEach(function() {
+    AppDispatcher = require('../../dispatcher/app_dispatcher');
+    MessageStore = require('../message_store');
+    callback = AppDispatcher.register.mock.calls[0][0];
+  });
+
+  it('creates a new message', function() {
+    callback(actionCreateMessage);
+    var messages = MessageStore.all();
+    expect(messages.length).toBe(3);
```

Comment:
let's just get the original length and verify that it goes up by 1, since the reason for the number 3 is nowhere in this file..